### PR TITLE
Plugin: Ensure that PHP code for blocks is correctly assigned to WP releases

### PR DESCRIPTION
**Pull request by gziolo**
## What?

Follow-up for https://github.com/WordPress/gutenberg/pull/36176.
Part of https://github.com/WordPress/gutenberg/issues/39889.

As part of this PR, I remove the usage of:

- `gutenberg_block_has_support`
- `gutenberg_migrate_old_typography_shape` filter's callback

They were added in WordPress 5.8.

I also moved block-related functionality to the appropriate compat layer files.

## Why?

To ensure that compatibility layer functionality gets removed whenever we bump the minimum version of WordPress supported in the Gutenberg plugin.

## How?

WordPres 5.9 compat:

- `_wp_normalize_relative_css_links` function
- `wp_enqueue_block_style` function
- `gutenberg_multiple_block_styles` filter's callback

WordPres 6.0 compat (needs to be backported to WP core):

- `wp_enqueue_block_view_script` function
- `gutenberg_block_type_metadata_multiple_view_scripts` filter's callback

## Testing Instructions

No functional changes. The plugin should still work as before.

### CODE REVIEW

File: lib/block-supports/border.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
  */
 function gutenberg_register_border_support( $block_type ) {
 	// Determine if any border related features are supported.
-	$has_border_support       = gutenberg_block_has_support( $block_type, array( '__experimentalBorder' ) );
+	$has_border_support       = block_has_support( $block_type, array( '__experimentalBorder' ) );
```
Comment (gziolo) [inline lib/block-supports/border.php:16]: https://developer.wordpress.org/reference/functions/block_has_support/ was introduced in WordPress 5.8 and that is the minimum supported version for the plugin.

File: lib/blocks.php [hunk 2]
```diff
@@ -397,299 +357,3 @@ function gutenberg_register_legacy_social_link_blocks() {
...
-	return true === $block_support || is_array( $block_support );
-}
-
-/**
- * Updates the shape of supports for declaring fontSize and lineHeight.
- *
- * @param array $metadata Metadata for registering a block type.
- * @return array          Metadata for registering a block type with the supports shape updated.
- */
-function gutenberg_migrate_old_typography_shape( $metadata ) {
```
Comment (gziolo) [inline lib/blocks.php:425]: Added to WordPress core in 5.8.0 release:
https://github.com/WordPress/wordpress-develop/blob/22fbb4a9ae877610f35fa5908c36839d9631674c/src/wp-includes/blocks.php#L1040-L1045

File: lib/blocks.php [hunk 3]
```diff
@@ -397,299 +357,3 @@ function gutenberg_register_legacy_social_link_blocks() {
...
-/**
- * Checks whether the current block type supports the feature requested.
- *
- * @param WP_Block_Type $block_type Block type to check for support.
- * @param array         $feature    Path of the feature to check support for.
- * @param mixed         $default    Fallback value for feature support, defaults to false.
- *
- * @return boolean                  Whether or not the feature is supported.
- */
-function gutenberg_block_has_support( $block_type, $feature, $default = false ) {
```
Comment (gziolo) [inline lib/blocks.php:410]: https://developer.wordpress.org/reference/functions/block_has_support/ was introduced in WordPress 5.8 and that is the minimum supported version for the plugin.

**Review by aristath — APPROVED** [review]
Nice! LGTM 👍

**Pull request merged**